conans/client/rest/uploader_downloader.py

```diff
@@ -19,16 +19,18 @@ def __init__(self, requester, output, verify, chunk_size=1000):
         self.verify = verify
 
     def upload(self, url, abs_path, auth=None, dedup=False, retry=1, retry_wait=0, headers=None):
+        # Send always the header with the Sha1
+        headers = headers or {}
+        headers["X-Checksum-Sha1"] = sha1sum(abs_path)
```

Comment:
`dedup` is configured by the server capabilities, that means SHA-1 will be included even when a server is not prepared to used it. Do you know if it could affect conan_server or Bintray?

Reply:
according to the HTTP specification, unknown headers should be just safely ignored